packages/flutter_client_sdk/lib/src/connection_manager.dart

```diff
@@ -160,26 +167,30 @@ final class ConnectionManager {
       return;
     }
 
-    /// Currently the foreground mode will always be whatever the last active
-    /// connection mode was.
+    // Currently the foreground mode will always be whatever the last active
+    // connection mode was.
     _destination.setMode(_currentConnectionMode);
     _destination.setEventSendingEnabled(true);
   }
 
   void _setBackgroundAvailableMode() {
+    // flush on backgrounding as application may be killed and we don't want to lose events.
+    _destination.flush();
+
     if (!_config.runInBackground) {
       // TODO: Can we support the backgroundDisabled data source status?
       // TODO: Is it acceptable for the data source status and `offline` to
       // report an `offline` status?
       _destination.setMode(ConnectionMode.offline);
-      _destination.setEventSendingEnabled(false);
+
+      // no need to flush here, we just did up above
+      _destination.setEventSendingEnabled(false, flush: false);
       return;
     }
 
-    /// If connections in the background are allowed, then use the same mode
-    /// as is configured for the foreground.
+    // If connections in the background are allowed, then use the same mode
+    // as is configured for the foreground.
     _setForegroundAvailableMode();
-    _destination.setEventSendingEnabled(true);
```

Comment:
Do not merge until bugfix for this line is merged separately to ensure changelog recording.